src/rpc/util.cpp

```diff
@@ -606,8 +606,9 @@ void RPCHelpMan::AppendArgMap(UniValue& arr) const
             map.push_back(m_name);
             map.push_back(i);
             map.push_back(arg_name);
-            map.push_back(arg.m_type == RPCArg::Type::STR ||
-                          arg.m_type == RPCArg::Type::STR_HEX);
+            // NOTE: push_back(bool) converts the bool to Number, so explicitly make a boolean UniValue first
```

Comment:
This note is only true for broken versions of univalue. push_back in our tree works just fine.